src/Symfony/Component/HttpClient/CurlHttpClient.php

```diff
@@ -436,12 +436,6 @@ private static function createRedirectResolver(array $options, string $host): \C
             $url = self::parseUrl(curl_getinfo($ch, \CURLINFO_EFFECTIVE_URL));
             $url = self::resolveUrl($location, $url);
 
-            curl_setopt($ch, \CURLOPT_PROXY, $options['proxy']
-                ?? ('https:' === $url['scheme'] ? $_SERVER['https_proxy'] ?? $_SERVER['HTTPS_PROXY'] ?? null : null)
```

Comment:
the reason why we have this code is to make the proxy URl-sensitive, so we should keep it and instead ensure the option is passed up tot this point

Reply:
I'm not sure, I'll let you check if there any diff in behavior :)